resolwe/flow/views/annotations.py

```diff
@@ -108,8 +128,27 @@ def create(self, request, *args, **kwargs):
             entity.has_permission(Permission.EDIT, request.user) for entity in entities
         ):
             raise exceptions.PermissionDenied()
-        self.perform_create(serializer)
+
+        # Create annotation values.
+        annotation_values = [AnnotationValue(**value) for value in validated_data]
+        self._validate_values(annotation_values)
+        inserted_values = self._bulk_create(annotation_values)
+
+        to_serialize = inserted_values if many else inserted_values[0]
+        serializer = self.get_serializer(to_serialize, many=many)
         headers = self.get_success_headers(serializer.data)
         return response.Response(
             serializer.data, status=status.HTTP_201_CREATED, headers=headers
         )
+
+    def put(self, request, *args, **kwargs):
+        """Update annotation value(s).
+
+        Create one or more annotation values. The request data can be either dict or
+        list (in case of bulk create).
+        """
+        # Call create, which calls _bulkcreate which will also allow update, since our
+        # method is PUT and not POST:
+        response = self.create(request, *args, **kwargs)
```

Comment:
Extract the create logic into a separate method and do not modify the response.

Reply:
OK.